src/types.ts

```diff
@@ -0,0 +1,50 @@
+export type SupportedDtype = 'Uint8' | 'Uint16' | 'Uint32' | 'Float32';
+export type SupportedTypedArray = InstanceType<
+  typeof globalThis[`${SupportedDtype}Array`]
+>;
+
+export interface LayerData {
+  data: SupportedTypedArray;
+  width: number;
+  height: number;
+}
+
+export type PixelSourceSelection<S extends string[]> = {
+  [K in S[number]]: number;
+};
+
+export interface RasterSelection<S extends string[]> {
+  selection: PixelSourceSelection<S>;
+}
+
+export interface TileSelection<S extends string[]> {
+  x: number;
+  y: number;
+  selection: PixelSourceSelection<S>;
+  signal?: AbortSignal;
+}
```

Comment:
I think we should probably remove the `signal` from this since it is not really part of the selection.    Relatedly, could we extend the `RasterSelection` interface here?

Reply:
Maybe `RasterOptions` and `TileOptions` are better names. I think it makes sense to have `signal` as an optional param on this interface, what would be the alternative?

Happy to extend `TileSelection` from `RasterSelection` but really it's just one shared key...

Reply:
I think what I'm trying to say is that `signal` is not really part of the selection, so we could make it a different argument to `getTile` for example.

Reply:
We could also just flatten the interfaces 

```typescript
interface PixelSource<S> {
  // ...
  getRaster(selection: S): Promise<LayerData>;
  getTile(x: number, y: number, selection: S, signal?: AbortSignal): Promise<LayerData>;
  // ...
}
```

Reply:
Yeah, I think I'd be ok with that - the API has been fairly stable with the only change really being the addition of a signal.

Reply:
Let's do this in a separate PR. The tests and source are working for now and the current interface mimic what we had prior. A subsequent PR would have a much smaller diff.

Reply:
Opened an issue.

Reply:
Getting back to this, if we keep with the convention of a selection basically containing its options, we'll need another one for volumes as well since that has its own parameters.  It feels like we should unify selections under one interface and then have an `opts` part of the signature for the functions.  This would also help unify the functions themselves.  I'll post some more information in the issue.